File: LeetCode_Algorithm/392(Is Subsequence)/392.js
var isSubsequence = function(s, t) {
  // sol1 O(t), O(1) 60~89ms 94~37%
  var sIdx = 0; var tIdx = 0;
  while (sIdx < s.length && tIdx < t.length) {
    if (s[sIdx] == t[tIdx]) {
      sIdx += 1; tIdx += 1; 
      continue 
    }
    tIdx += 1
  }
  return (sIdx == s.length)
};
console.log(isSubsequence("abc", "acbac"))

File: LeetCode_Algorithm/392(Is Subsequence)/392.py
class Solution:
    def isSubsequence(self, s: str, t: str) -> bool:
        # sol1 O(m), O(1) 44~50ms 57~41%
        # m is length of target string
        sIdx = 0; tIdx = 0
        while sIdx < len(s) and tIdx < len(t):
            if s[sIdx] == t[tIdx]: 
                sIdx += 1; tIdx += 1
            elif s[sIdx] != t[tIdx]:
                tIdx += 1
        return sIdx == len(s)


print(Solution().isSubsequence("abc", "ahbgdc"))